enterprise/server/usage/usage.go

```diff
@@ -136,3 +375,40 @@ func countsToMap(tu *tables.UsageCounts) (map[string]int64, error) {
 	}
 	return counts, nil
 }
+
+// stringMapToCounts converts a Redis hashmap containing usage counts to
+// tables.UsageCounts.
+func stringMapToCounts(h map[string]string) (*tables.UsageCounts, error) {
+	hInt64 := map[string]int64{}
+	for k, v := range h {
+		count, err := strconv.ParseInt(v, 10, 64)
+		if err != nil {
+			return nil, status.InvalidArgumentErrorf("Invalid usage count in Redis hash: %q => %q", k, v)
+		}
+		hInt64[k] = count
+	}
+	return &tables.UsageCounts{
+		Invocations:            hInt64["invocations"],
+		CASCacheHits:           hInt64["cas_cache_hits"],
+		ActionCacheHits:        hInt64["action_cache_hits"],
+		TotalDownloadSizeBytes: hInt64["total_download_size_bytes"],
+	}, nil
+}
+
+// usageUpdates returns a GORM updates map that finalizes the usage row up to
```

Comment:
lol, can we un-gormify this and do this with SQL?

Reply:
Done